### PR TITLE
update the reference DB section

### PR DESCRIPTION
**Pull request by eparejatobes**
Since the preprint this has significantly changed (and improved!). We need to

- [x] describe (and cite) the original data source, RNACentral
- [x] explain what kind of sequences are considered 16S
- [x] how redundant assignments are collapsed
- [x] our strategy for identifying and discarding wrong assignments
- [x] how all this can be re-used for any subset of RNACentral, like for example 18S

### CODE REVIEW

**Comment by eparejatobes** [comment]
See ohnosequences/db.rna16s#32

**Comment by eparejatobes** [comment]
Please @marina-manrique @rtobes check this. I do think we should a figure for what we consider a wrong assignment, and how are we discarding them.

File: docs/mg7-paper/paper.md [hunk 1]
```diff
@@ -121,11 +121,25 @@ Bio4j [@pareja2015bio4j] is a data platform integrating data from different reso
...
-Our 16S database building strategy allows the substitution of the 16S database by any other subset of **nt**, even by the complete **nt** database if it would be needed, for example, for analyzing shotgun metagenomics data. This possibility of changing the reference database provides flexibility to the system enabling it for easy updating and project-driven personalization.
+1. are annotated as being of `rRNA` type^[We are aware of the existence of a gene annotation corresponding to 16S in RNACentra, that we are **not using** due to a significant amount of 16S sequences lacking it]
+2. their length is at least `1300`^[Note that 16S sequences are sometimes part of an entry corresponding to whole small subunits; that's why we do not set a maximum length threshold]
+3. have at least one taxonomic assignment to a descendant of *Bacteria* or *Archaea*
+4. their lineage does not contain a set of taxa deemed uninformative^[for example: "unclassified Bacteria (miscellaneous)" or "unclassified". Assigning a read to an "unclassified" taxon defeats the first and foremost goal of taxonomic profiling: *classification*.]
```
Comment (laughedelic) [inline docs/mg7-paper/paper.md]: Isn't it too many footnotes for one paragraph? Except the first one, why not to just include this text here?

Comment (eparejatobes) [inline docs/mg7-paper/paper.md]: As I view it, they should be footnotes if they are semantically footnotes; if a paragraph ends up having a lot of them, so be it.

**Comment by rtobes** [comment]
I am working in the section **16S-DB7 Reference Database Construction** because is not easy to understand. I am preparing a figure about it.

File: docs/mg7-paper/paper.md [hunk 2]
```diff
@@ -121,11 +121,25 @@ Bio4j [@pareja2015bio4j] is a data platform integrating data from different reso
...
+
```
Comment (rtobes) [inline docs/mg7-paper/paper.md]: I don´t understand the sentence in 131 line

File: docs/mg7-paper/paper.md [hunk 3]
```diff
@@ -121,11 +121,25 @@ Bio4j [@pareja2015bio4j] is a data platform integrating data from different reso
...
+After this first step, we drop redundant assignments: if sequences $S_1 \subseteq S_2$ share an assignment $T$, it gets dropped from $S_1$; sequences which as a result of this process end up having no assignments are removed.
+
+Once we have non-redundant set of RNA sequences containing all those corresponding to 16S, we can apply our procedure for checking internal consistency of taxonomic assignments. We run MG7 using this set of sequences as query, with reference all but the one we are trying to assign. Under the (reasonable) assumption of 16S sequence similarity being correlated with the taxonomy tree topology, the MG7 result should either
```
Comment (rtobes) [inline docs/mg7-paper/paper.md]: Once we have non-redundant set of ribosomal RNA sequences (containing all those corresponding to 16S)

File: docs/mg7-paper/paper.md [hunk 4]
```diff
@@ -121,11 +121,25 @@ Bio4j [@pareja2015bio4j] is a data platform integrating data from different reso
...
+
+1. be a taxon *close* to the original assignment, if there are sequences similar enough in the reference database
+2. be empty, due to this sequence having no similar sequences in the reference database
+
+In any other case, under our assumptions, this assignment should be discarded. What we consider as close is: the parent of the MG7 assignment should be contained in the lineage of the original assignment.
+
```
Comment (rtobes) [inline docs/mg7-paper/paper.md]: Remove from 138 line: 

> What we consider as close is: the parent of the MG7 assignment should be contained in the lineage of the original assignment.

File: docs/mg7-paper/paper.md [hunk 5]
```diff
@@ -121,11 +121,25 @@ Bio4j [@pareja2015bio4j] is a data platform integrating data from different reso
...
+4. their lineage does not contain a set of taxa deemed uninformative^[for example: "unclassified Bacteria (miscellaneous)" or "unclassified". Assigning a read to an "unclassified" taxon defeats the first and foremost goal of taxonomic profiling: *classification*.]
+
+After this first step, we drop redundant assignments: if sequences $S_1 \subseteq S_2$ share an assignment $T$, it gets dropped from $S_1$; sequences which as a result of this process end up having no assignments are removed.
+
+Once we have non-redundant set of RNA sequences containing all those corresponding to 16S, we can apply our procedure for checking internal consistency of taxonomic assignments. We run MG7 using this set of sequences as query, with reference all but the one we are trying to assign. Under the (reasonable) assumption of 16S sequence similarity being correlated with the taxonomy tree topology, the MG7 result should either
+
+1. be a taxon *close* to the original assignment, if there are sequences similar enough in the reference database
+2. be empty, due to this sequence having no similar sequences in the reference database
+
```
Comment (rtobes) [inline docs/mg7-paper/paper.md]: 1.  be a taxon *close* to the original assignment, if there are sequences similar enough in the reference database. [We consider close as: the parent of the MG7 assignment should be contained in the lineage of the original assignment]
2. be empty, due to this sequence having no similar sequences in the reference database and it is not possible to do a taxonomic assignment for it

File: docs/mg7-paper/paper.md (hunk 3 above)
Comment (rtobes) [inline docs/mg7-paper/paper.md]: Under the (reasonable) assumption of 16S sequence similarity being correlated with the taxonomy tree topology, the MG7 taxonomic assignment result for a DB sequence should either:

File: docs/mg7-paper/paper.md (hunk 2 above)
Comment (eparejatobes) [inline docs/mg7-paper/paper.md]: In the pdf this looks like this: S1 ⊆ S2. We are removing all redundant assignments, and then sequences which as a result have no assignments left.

**Comment by eparejatobes** [comment]
@rtobes fixed everything in 094deab, thank you!

**Comment by eparejatobes** [comment]
If everyone's OK, I'd rather merge this and do the figure in a separate branch.

File: docs/mg7-paper/paper.md [hunk 6]
```diff
@@ -121,11 +121,25 @@ Bio4j [@pareja2015bio4j] is a data platform integrating data from different reso
...
 
-Certainly, if metagenomics results are easily integrated with the theoretical and experimental knowledge of each specific area, the impact of metagenomics will be higher than if it progresses as a disconnected research branch. Considering that metagenomics data interoperability, which is especially critical in clinical environments, requires a stable taxonomy to be used as reference, we decided to rely on the most widely used taxonomy: the NCBI taxonomy. In addition, the biggest global sequence database GenBank follows this taxonomy to register the origin of all their submitted sequences.
-Our 16S database building strategy allows the substitution of the 16S database by any other subset of **nt**, even by the complete **nt** database if it would be needed, for example, for analyzing shotgun metagenomics data. This possibility of changing the reference database provides flexibility to the system enabling it for easy updating and project-driven personalization.
+1. are annotated as being of `rRNA` type^[We are aware of the existence of a gene annotation corresponding to 16S in RNACentra, that we are **not using** due to a significant amount of 16S sequences lacking it]
```
Comment (rtobes) [inline docs/mg7-paper/paper.md]: RNACentral

**Comment by rtobes** [comment]
OK for a new branch for the figure about the DB construction

File: docs/mg7-paper/paper.md (hunk 6 above)
Comment (laughedelic) [inline docs/mg7-paper/paper.md]: And on their website it's written "RNAcentral"

**Pull request merged**